apps/src/gamelab/actions.js

```diff
@@ -41,3 +49,103 @@ module.exports.setInitialLevelProps = function (props) {
     props: props
   };
 };
+
+/**
+ * Push full animation metadata into the store, usually on first load
+ * from the sources API.
+ * 
+ * @param {Object} metadata
+ * @returns {{type: ActionType, metadata: Object}}
+ */
+module.exports.setInitialAnimationMetadata = function (metadata) {
+  return {
+    type: ActionType.SET_INITIAL_ANIMATION_METADATA,
+    metadata: metadata
+  }
+};
+
+module.exports.addAnimation = function (animationProps) {
+  // TODO: Validate animationProps?
+  return function(dispatch, getState) {
+    dispatch({
+      type: ActionType.ADD_ANIMATION_AT,
+      index: getState().animations.length,
+      animationProps: animationProps
+    });
+  }
+};
+
+module.exports.cloneAnimation = function (animationKey) {
+  return function(dispatch, getState) {
```

Comment:
This is an example of a redux-thunk enabled action creator, to do the async 'clone' operation.